build/factory/memory-server-factory.js

```diff
@@ -0,0 +1,108 @@
+exports.MemoryServerFactory = void 0;
+var mongodb_1 = require("mongodb");
+var MongoMemoryServer_1 = require("mongodb-memory-server-core/lib/MongoMemoryServer");
+var MemoryServerFactory = /** @class */ (function () {
+    function MemoryServerFactory() {
+    }
+    MemoryServerFactory.generateMemoryServer = function (dbName, port, storageEngine, version) {
+        return __awaiter(this, void 0, void 0, function () {
+            var instanceOpts, binaryOpts, opts, server;
+            return __generator(this, function (_a) {
+                switch (_a.label) {
+                    case 0:
+                        instanceOpts = {};
+                        binaryOpts = { checkMD5: true };
+                        if (dbName) {
+                            instanceOpts.dbName = dbName;
+                        }
+                        if (port) {
+                            instanceOpts.port = port;
+                        }
+                        if (storageEngine) {
+                            instanceOpts.storageEngine = storageEngine;
+                        }
+                        if (version) {
+                            binaryOpts.version = version;
+                        }
+                        opts = {
+                            instance: instanceOpts,
+                            binary: binaryOpts,
+                        };
+                        return [4 /*yield*/, MongoMemoryServer_1.MongoMemoryServer.create(opts)];
+                    case 1:
+                        server = _a.sent();
+                        return [2 /*return*/, server];
+                }
+            });
+        });
+    };
+    MemoryServerFactory.verifyMemoryServer = function (server) {
+        return __awaiter(this, void 0, void 0, function () {
+            var client, memoryServerUri, err_1;
+            return __generator(this, function (_a) {
+                switch (_a.label) {
+                    case 0:
+                        _a.trys.push([0, 2, 3, 6]);
+                        memoryServerUri = server.getUri();
+                        client = new mongodb_1.MongoClient(memoryServerUri, { useUnifiedTopology: true, useNewUrlParser: true });
+                        return [4 /*yield*/, client.connect()];
+                    case 1:
+                        _a.sent();
+                        return [2 /*return*/, memoryServerUri];
+                    case 2:
+                        err_1 = _a.sent();
+                        console.error(err_1);
+                        throw err_1;
+                    case 3:
+                        if (!(client === null || client === void 0 ? void 0 : client.isConnected())) return [3 /*break*/, 5];
+                        return [4 /*yield*/, client.close()];
+                    case 4:
+                        _a.sent();
+                        _a.label = 5;
+                    case 5: return [7 /*endfinally*/];
+                    case 6: return [2 /*return*/];
```

Comment:
As well as here throughout the function. I'm not 100% certain as to what would be a better option here to handle this instead as I see you're just returning numbers.